knox/views.py

```diff
@@ -79,3 +79,33 @@ def post(self, request, format=None):
         user_logged_out.send(sender=request.user.__class__,
                              request=request, user=request.user)
         return Response(None, status=status.HTTP_204_NO_CONTENT)
+
+
+class TokenRefreshView(APIView):
+
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request, format=None):
+        return Response(
+            {'expiry': request.auth.expiry},
+            status=status.HTTP_200_OK
+        )
+
+
+class TokenRefreshAllView(APIView):
+
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request, format=None):
+        request.user.auth_token_set.all().update(
```

Comment:
This should probably be filtered so that we dont make expired tokens valid and non expired again.

Reply:
It shouldn't happen as the cleanup phase is triggered during the authentication phase for all token of the identified user:

https://github.com/James1345/django-rest-knox/blob/c1972fcddf86179680522d6af1b463b02c49b00c/knox/auth.py#L58-L70

https://github.com/James1345/django-rest-knox/blob/c1972fcddf86179680522d6af1b463b02c49b00c/knox/auth.py#L100-L115

Reply:
could you describe a use case where one client needs to refresh all its token? i would think they are not used in one device and therefore it does not make sense to me to extend the tokens of non-active devices.

Reply:
@belugame I cannot, but it was mentioned in the discussion thread, I know I thought it was a good idea but now I am not so sure anymore because the more I think of it, the less convinced I become. It should be up to each individual client to do this, not the responsability of one client to do it for everybody else.

@johnraz what do you think about this? Should we nuke this feature?

Reply:
Yea I realize as well that this is more than a niche feature. I can't even remember the precise use-case I had in mind, I think I was confused by the logout all use-case which makes perfect sense... Nuke! 💥